### ISSUE TITLE
Perfdata shown in Livestatus, but not being written to rrd file

### ISSUE THREAD

**Issue by ianling** [naparuba/shinken#1734]
Here's the perfdata in question:

    'Net Silicon NS9xxx Ethernet: Rev 1.3-ifInOctets'=16897.5992702;;;; 'Net Silicon NS9xxx Ethernet: Rev 1.3-ifOutOctets'=18414.3442808;;;; 'Net Silicon NS9xxx Ethernet: Rev 1.3-ifInErrors'=0.0;;;; 'Net Silicon NS9xxx Ethernet: Rev 1.3-ifOutErrors'=0.0;;;; 'Net Silicon NS9xxx Ethernet: Rev 1.3-ifInNUcastPkts'=0.0;;;; 'Net Silicon NS9xxx Ethernet: Rev 1.3-ifOutNUcastPkts'=0.0;;;; 'Net Silicon NS9xxx Ethernet: Rev 1.3-ifInUcastPkts'=95.3885140996;;;; 'Net Silicon NS9xxx Ethernet: Rev 1.3-ifOutUcastPkts'=185.01893164;;;; 'Ethernet (No ip)-ifInOctets'=109819723.987;;;; 'Ethernet (No ip)-ifOutOctets'=140057423.12;;;; 'Ethernet (No ip)-ifInErrors'=0.262363516275;;;; 'Ethernet (No ip)-ifOutErrors'=0.0;;;; 'Ethernet (No ip)-ifInNUcastPkts'=4642.92115823;;;; 'Ethernet (No ip)-ifOutNUcastPkts'=8189.02068675;;;; 'Ethernet (No ip)-ifInUcastPkts'=8736443.54406;;;; 'Ethernet (No ip)-ifOutUcastPkts'=11978892.8948;;;; 'Ethernet (No ip)-ifInOctets'=17844603706.4;;;; 'Ethernet (No ip)-ifOutOctets'=1.94509311331e+11;;;; 'Ethernet (No ip)-ifInErrors'=1.86404375935e+11;;;; 'Ethernet (No ip)-ifOutErrors'=0.0;;;; 'Ethernet (No ip)-ifInNUcastPkts'=1960216.30429;;;; 'Ethernet (No ip)-ifOutNUcastPkts'=1.9194204649e+11;;;; 'Ethernet (No ip)-ifInUcastPkts'=1183655640.95;;;; 'Ethernet (No ip)-ifOutUcastPkts'=2.01242944778e+11;;;; 'Ethernet (No ip)-ifInOctets'=1.4430095472e+11;;;; 'Ethernet (No ip)-ifOutOctets'=1.53592423055e+11;;;; 'Ethernet (No ip)-ifInErrors'=0.0;;;; 'Ethernet (No ip)-ifOutErrors'=0.0;;;; 'Ethernet (No ip)-ifInNUcastPkts'=2.0744751988e+11;;;; 'Ethernet (No ip)-ifOutNUcastPkts'=2.09186961673e+11;;;; 'Ethernet (No ip)-ifInUcastPkts'=2.01888328535e+11;;;; 'Ethernet (No ip)-ifOutUcastPkts'=2.04316200874e+11;;;;

I have a service, 'serviceSNMP', that pulls interface data from 'hostA' via SNMP, as seen above. The service works perfectly fine, and when I refresh the web UI, I can see that the perfdata is updating (i.e., the numbers change, so it is definitely pulling new data from the host).

However, the PNP4Nagios graphs do not display any data, so I checked the /usr/local/pnp4nagios/var/perfdata folder, which is where my rrd's and xml's are contained. There is no rrd file or xml file for serviceSNMP, but there is for the hostA's host check perfdata (ping response times, packet loss). There are rrd and xml files for serviceSNMP on other hosts (hostB, hostC, etc.) though, so it is not a problem with serviceSNMP itself.

Additionally, there is a second service on hostA, serviceNumberTwo, and that service is working perfectly, INCLUDING writing to a rrd file, so it is not a problem with hostA. It appears that there is some issue with the way that the perfdata is handled from serviceSNMP on this hostA.

To summarize:

* serviceSNMP writes rrd files for hostB, hostC, and so on
* serviceNumberTwo writes rrd files for hostA
* hostA's rrd files for its host checks are written perfectly.
* The problem: serviceSNMP does not write rrd files for hostA.

 My guesses:

1. The data is too large. For example: `'Ethernet (No ip)-ifInErrors'=1.86404375935e+11` 1.86e+11 is a pretty big number, is it causing some sort of overflow?

2. The titles of the data are also large. For example: `Net Silicon NS9xxx Ethernet: Rev 1.3-ifOutNUcastPkts` is a long title, could that cause overflow?

3. The data is sometimes in scientific notation (see #1 above). Can scientific notation be written to an rrd file?

Any ideas?

**Comment by Corbyn** [comment]
Hi,

seems like this is a PNP4Nagios Problem. First step would be to turn on PNP4Nagios debug logging...
process_perfdata.cfg:
LOG_LEVEL = debug

PNP4Nagios is very stringent concerning the format of perfdata. Perhaps the "e+" notation is just not supported.

regards
